inference/enterprise/workflows/complier/steps_executors/models.py

```diff
@@ -1279,3 +1282,62 @@ def detect_qr_codes(
             }
         )
     return predictions
+
+
+async def run_barcode_detection_step(
+    step: BarcodeDetection,
+    runtime_parameters: Dict[str, Any],
+    outputs_lookup: OutputsLookup,
+    model_manager: ModelManager,
+    api_key: Optional[str],
+    step_execution_mode: StepExecutionMode,
+) -> Tuple[NextStepReference, OutputsLookup]:
+    image = get_image(
+        step=step,
+        runtime_parameters=runtime_parameters,
+        outputs_lookup=outputs_lookup,
+    )
+    decoded_images = [load_image(e)[0] for e in image]
+    image_metadata = [
+        {"width": img.shape[1], "height": img.shape[0]} for img in decoded_images
+    ]
+    image_parent_ids = [img["parent_id"] for img in image]
+    predictions = [
+        detect_barcodes(image=image, parent_id=parent_id)
+        for image, parent_id in zip(decoded_images, image_parent_ids)
+    ]
+
+    outputs_lookup[construct_step_selector(step_name=step.name)] = {
+        "parent_id": image_parent_ids,
+        "predictions": predictions,
+        "image": image_metadata,
+        "prediction_type": "barcode-detection",
+    }
+    return None, outputs_lookup
+
+
+def detect_barcodes(
+    image: np.ndarray, parent_id: str
+) -> Dict[str, Union[str, np.ndarray]]:
+    barcodes = zxingcpp.read_barcodes(image)
+    predictions = []
+
+    for barcode in barcodes:
+        width = barcode.position.top_right.x - barcode.position.top_left.x
+        height = barcode.position.bottom_left.y - barcode.position.top_left.y
+
+        predictions.append(
+            {
+                "parent_id": parent_id,
+                "class": "barcode",
+                "class_id": 0,
+                "confidence": 1.0,
+                "x": int(math.floor(barcode.position.top_left.x + width / 2)),
```

Comment:
Please note: `top_left` refers to the top-left corner of the barcode symbol. So if you have a QRCode that is scanned upside-down, then this would be the bottom right corner in your image. Hence your `x,y` would be outside the symbol, not in the center. A better approximation of the center would be the average of the `top_left` and `bottom_right` points.

Reply:
Addendum: I just realized that `x,y` are indeed calculated properly in that particular case (rotated 180°) but in that case your values for `width` and `height` will be negative.